packages/react-ui-core/src/Counter/Counter.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+export default class Counter extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      count: this.props.count,
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+  handleClick(count) {
+    this.props.handleClick(count)
+    this.setState({
+      count,
+    })
+  }
+  render() {
+    const { count } = this.state
+    const { theme, label, leftUnit, rightUnit, id } = this.props
+    return (
+      <div id={id} className={theme.container}>
+        <div className={theme.label}>
+          {label}
+        </div>
+        <div className={theme.componentContainer}>
+          <span
+            onClick={() => this.handleClick(this.state.count + 1)}
+            className={classNames('leftUnit', theme.contentInnards, theme.shape)}
+          >
+            {this.props.leftOperator}
+          </span>
+          <span
+            className={classNames(theme.contentInnards, theme.count)}
+          >{`${leftUnit} ${count} ${rightUnit}`}</span>
+          <span
+            onClick={() => this.handleClick(this.state.count - 1)}
+            className={classNames('rightUnit', theme.contentInnards, theme.shape)}
+          >
+            {this.props.rightOperator}
+          </span>
+        </div>
+      </div>
+    )
+  }
+}
+
+Counter.defaultProps = {
+  theme: {},
+  count: 1,
+  leftUnit: '',
+  rightUnit: '',
+  handleClick: () => {
+    alert('please add rightOperatorClick handler function')
+  },
+  leftOperator: <span>-</span>,
+  righttOperator: <span>+</span>,
+}
+
+Counter.propTypes = {
+  leftOperator: PropTypes.element,
+  righttOperator: PropTypes.element,
```

Comment:
I don't think we should label things `leftOperator, `righttOperator` (which is spelled wrong here as well), `leftUnit` or `rightUnit`.  If the increment / decrement buttons are on the top and bottom these names would be irrelevant.  I think naming them `increment` or `decrement` or something would be better.